src/main/java/com/rabbitmq/client/impl/nio/NioLoop.java

```diff
@@ -276,17 +273,7 @@ protected void dispatchIoErrorToConnection(final SocketChannelFrameHandlerState
         // In case of recovery after the shutdown,
         // the new connection shouldn't be initialized in
         // the NIO thread, to avoid a deadlock.
-        Runnable shutdown = new Runnable() {
-
-            @Override
-            public void run() {
-                try {
-                    state.getConnection().handleIoError(ex);
-                } catch(AssertionError e) {
-                    LOGGER.warn("Assertion error during error dispatching to connection: " + e.getMessage());
-                }
-            }
-        };
+        Runnable shutdown = () -> state.getConnection().handleIoError(ex);
```

Comment:
Is it time to switch to lambdas and thus require JDK 8? Will Spring-AMQP require JDK 8 in 2.0?